extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/SecurityIdentityHandler.java

```diff
@@ -0,0 +1,31 @@
+package io.quarkus.amazon.lambda.http;
+
+import java.util.Map;
+
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.vertx.http.runtime.QuarkusHttpHeaders;
+import io.quarkus.vertx.http.runtime.security.QuarkusHttpUser;
+import io.vertx.core.Handler;
+import io.vertx.core.MultiMap;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * Filter that will check to see if LambdaHttpHandler forwarded a SecurityIdentity.
+ * If so, it sets the RoutingContext.user
+ *
+ */
+public class SecurityIdentityHandler implements Handler<RoutingContext> {
```

Comment:
This should be a `io.quarkus.vertx.http.runtime.security.HttpAuthenticationMechanism`, it should create a 'LambdaAuthenticationRequest` (or whatever you want to call it), and ask the `IdentityProviderManager` to authenticate it.

Reply:
I'll check it out, but I hope its as easy to write as my current implementation.  IIRC, the whole HttpAuthenticationMechanism took a bit to grok and what I currently have is pretty simple.

Reply:
Pretty simple for users to implement I mean.  I don't care how hard it is for me to implement :)

Reply:
The mechanism should just take the request object you stashed, wrap it in an AuthenticationRequest subclass, try and authenticate the request. Mostly it should be very very similar to the code you already have, just in different places.